microservices/nris_api/backend/app/nris/scheduled_jobs/nris_jobs.py

```diff
@@ -11,19 +11,21 @@
 
 def _schedule_nris_etl_jobs(app):
     # the schedule of these jobs is set using server time (UTC)
-    app.apscheduler.add_job(func=nris_etl_job,
-                            trigger='cron', id='ETL', hour=11, minute=0)
+    app.apscheduler.add_job(func=nris_etl_job, trigger='cron', id='ETL', hour=11, minute=0)
 
 
 def nris_etl_job():
     """This nightly job initiates the ETL from NRIS into our app domain."""
 
     job_running = cache.get(NRIS_JOB_PREFIX + NRIS_ETL_JOB)
     if job_running is None:
-        cache.set(NRIS_JOB_PREFIX + NRIS_ETL_JOB,
-                  'True', timeout=TIMEOUT_12_HOURS)
-        _run_nris_etl()
-        cache.delete(NRIS_JOB_PREFIX + NRIS_ETL_JOB)
+        try:
+            cache.set(NRIS_JOB_PREFIX + NRIS_ETL_JOB, 'True', timeout=TIMEOUT_12_HOURS)
+            _run_nris_etl()
+        finally:
```

Comment:
Using a finally is probably the right approach here, really caused us some heartache with debugging otherwise.